fix(service): twenty template

yanniksuess opened the pull request:
## Changes

- feat(service): improve twenty template
  - added worker service for background job processing (email, webhooks, cron
jobs)
  - upgraded to postgres 16
  - updated some envs which fix some issues
  - added volumes to persist uploads

yanniksuess committed:
Set workspace inactive days for notifications and deletion

Fix workspace inactive days validation error

Initialize both WORKSPACE_INACTIVE_DAYS_BEFORE_DELETION (21) and WORKSPACE_INACTIVE_DAYS_BEFORE_NOTIFICATION (7) to prevent validation error requiring notification days to be less than deletion days.

yanniksuess committed:
Merge branch 'next' into patch-1

ShadowArcanist commented on templates/compose/twenty.yaml:
```diff
@@ -31,8 +31,8 @@ services:
-      - WORKSPACE_INACTIVE_DAYS_BEFORE_NOTIFICATION=$WORKSPACE_INACTIVE_DAYS_BEFORE_NOTIFICATION
-      - WORKSPACE_INACTIVE_DAYS_BEFORE_DELETION=$WORKSPACE_INACTIVE_DAYS_BEFORE_DELETION
+      - WORKSPACE_INACTIVE_DAYS_BEFORE_NOTIFICATION=7
+      - WORKSPACE_INACTIVE_DAYS_BEFORE_DELETION=21
```
```suggestion
      - WORKSPACE_INACTIVE_DAYS_BEFORE_NOTIFICATION=${WORKSPACE_INACTIVE_DAYS_BEFORE_NOTIFICATION:-7}
      - WORKSPACE_INACTIVE_DAYS_BEFORE_DELETION=${WORKSPACE_INACTIVE_DAYS_BEFORE_DELETION:-21}
```

Please don't hardcode these values, allow users to change the value from the UI

ShadowArcanist reviewed (changes requested):
Please implement the suggested changes and rewrite the PR title to specify the service name to avoid confusion (you can check other service related PRs for an inspiration)

yanniksuess committed:
Update templates/compose/twenty.yaml

Better approach 👌

Co-authored-by: ShadowArcanist <162910371+ShadowArcanist@users.noreply.github.com>

yanniksuess committed:
Merge branch 'next' into patch-1

yanniksuess committed:
Merge branch 'next' into patch-1

yanniksuess committed:
Refactor twenty.yaml for improved configuration

- Added storage for image and other uploads
- Added worker for maintenance
- Added health check retries for the worker

yanniksuess committed:
Add metadata comments to twenty.yaml

Missing comments

yanniksuess commented:
Fix image upload 404 errors by adding persistent storage and worker service

  Changes:
  - Added volume mount
  `twenty-local-storage:/app/packages/twenty-server/.local-storage` to
  persist uploaded files across container restarts
  - Added worker service for background job processing (email, webhooks, cron
   jobs)
  - Worker has all environment variables from main service +
  DISABLE_DB_MIGRATIONS and DISABLE_CRON_JOBS_REGISTRATION
  - Increased healthcheck retries from 15 to 60 to allow more startup time
  - Worker depends on twenty service health check
  - Upgraded to Postgres 16 (Since version [v0.43.0](https://twenty.com/developers/section/self-hosting/upgrade-guide#v0.42.0-to-v0.43.0))

  This fixes the 404 errors when uploading images - files are now saved to a
  persistent Docker volume instead of ephemeral container storage.

peaklabs-dev committed:
Merge branch 'next' into patch-1

peaklabs-dev committed:
refactor(service): improve twenty

- pin twenty version
- autogenerate postgres user
- fix formatting, naming and order

peaklabs-dev reviewed (changes requested):
~~I tested the template and it is not working properly for me. Could you please test it again and apply any necessary fixes.~~

peaklabs-dev committed:
fix: twenty startup time and broken env

peaklabs-dev reviewed (approved):
Found the issue. Thanks for the PR.
